client/landing/stepper/declarative-flow/internals/steps-repository/unified-plans/index.tsx

```diff
@@ -80,6 +92,39 @@ export default function PlansStepAdaptor( props: StepProps ) {
 		setPlanInterval( intervalType );
 	};
 
+	/**
+	 *  Plan step switches the selected theme to default twentytwentyfour when the plan is free
+	 *  but the selected design requires a paid plan.
+	 */
+	const switchPaidDesignToDefault = ( stepInfo: ProvidedDependencies ) => {
+		const hasPaidDesign = Boolean(
+			selectedDesign?.design_tier && selectedDesign?.design_tier !== 'free'
+		);
+		const isOnboarding = isOnboardingFlow( props.flow ) && isGoalFirstExperiment;
+
+		if ( ! hasPaidDesign || !! stepInfo.cartItems || ! isOnboarding ) {
+			return;
+		}
+
+		if ( site ) {
+			setPendingAction( async () => {
+				return setDesignOnSite( site?.ID, defaultDesign, {
+					styleVariation: defaultDesign?.style_variations?.[ 0 ],
+				} ).then( async ( theme: ActiveTheme ) => {
+					await updateLaunchpadSettings( site?.ID || '', {
+						checklist_statuses: { design_completed: false },
+					} );
+					return reduxDispatch( setActiveTheme( site?.ID || -1, theme ) );
+				} );
+			} );
+		}
+		setSelectedDesign( { ...defaultDesign, default: true } );
+
+		recordTracksEvent( 'calypso_paid_theme_auto_switch', {
```

Comment:
This looks like a new event. Are we planning to register it and maybe share it here (pfYzsZ-16M-p2) so people will know it exists?

Reply:
Yeah we will register the event